homeassistant/components/usage_prediction/common_control.py

```diff
@@ -93,22 +93,30 @@ async def async_predict_common_control(
     Args:
         hass: Home Assistant instance
         user_id: User ID to filter events by.
-
-    Returns:
-        Dictionary with time categories as keys and lists of most common entity IDs as values
     """
     # Get the recorder instance to ensure it's ready
     recorder = get_instance(hass)
     ent_reg = er.async_get(hass)
 
     # Execute the database operation in the recorder's executor
     return await recorder.async_add_executor_job(
```

Comment:
Would it be cleaner to return the result of session.connection().execute(query).all() and than process it here to avoid fetching everything from the state machine ?